scripts/auto_backtester/backtester.js

```diff
@@ -16,11 +16,12 @@ let parallel  = require('run-parallel-limit');
 let json2csv  = require('json2csv');
 let roundp    = require('round-precision');
 let fs        = require('fs');
+let path = require('path')
 let StripAnsi = require('strip-ansi');
 
 let VERSION = 'Zenbot 4.04 Backtester v0.2';
 
-let PARALLEL_LIMIT = require('os').cpus().length;
+let PARALLEL_LIMIT = 3//require('os').cpus().length;
```

Comment:
why not make this depending on the cpus?

Reply:
If I have to take a guess this was introduced for debug purposes and then forgotten to undo

Reply:
correct.   it wasn't supposed to be checked in.  Still figuring out git.  totally breaks my source control flow ( used to SVN )

Reply:
@station384 since you are new to git, I assume you are also a bit new to github, like me. Just sharing some info that I figured out also in one of my first pull requests... what I did is to merge changes from the upstream branch to my fork, commit one change and finally do a pull request comparing my fork with the official repo/branch. As I noted here, the pull request included lot of commits I didn't do. Somehow this is a weird behavior of github.

To avoid that (next time, now too late since it was merged already), at best you keep working on your branch, but just a few minutes before making the pull request you create a new local branch based on the upstream one (I try to call such branches with the name of what they do, or issue id... but the name doesn't matter). Then you cherry pick your changes or merge and make just one commit, and push your branch. Then, when you do the pull request, github will compare that new branch with your changes to the actual one, where you want your changes to go. In such a case then it'll pick only your changes and nothing else added in the meantime by other developers. I had the same issue, if you see your commit now it seems you did so many changes, but this is because github included lot of stuff really unrelated to your change...

@DeviaVir I think that this should be documented, if not in the readme, there should be a readme for contributors with guidelines to follow.

I see there is a code of conduct, maybe it makes sense to create a DEVELOPMENT.md file or something similar, containing also such hints as the one I wrote above, useful also for people new to github (this issue isn't really documented in github)...

Reply:
@firepol usually one would add a `CONTRIBUTING.md` but I don't think you'll find basic git usage in that document, usually. Would not be a bad place for it though.

The reason we don't have a `CONTRIBUTING.md` is because it's really a lot of writing and setting up general rules, which I simply haven't had the time yet to do. When I do have a bit of spare time I prefer to review PR's and play around with strats/scripts/ideas vs. writing up documentation, if you know what I mean.

If anyone reading this feels inclined to jump in and write this document, the guidelines are available here:
https://help.github.com/articles/setting-guidelines-for-repository-contributors/

Reply:
I sort of figured out the branching part, but wasn't sure, for every change I want to push I was making branches off of my fork of unstable,  the largest problem I've had is when I've committed something, and published.  in SVN I would just check out from a prior check in and recheck in.  I assume GIT is similar.  

I also get messed up because in svn, I would just check in on, and then there is a separate commit where I can cherry pick the files I commit (I usually have 4 or more mods going at once in the same branch in SVN). 
My SVN process seems to be what the branching in git does.

Git has more planning ahead of what you want to commit, where as SVN happens later.

Sooner or later I'll catch on, I've been using SVN for the last decade so old habits die hard, and the stuff I learned on git from about 2 years ago has to be relearned :/  

But thanks for the tips,   confirms a couple of things, and lay off the batch commit of files like I have been doing and start doing them very specifically as you described.

and "the pull request included lot of commits I didn't do. Somehow this is a weird behavior of github"  this was bit alarming, to the point of me redoing branches and such.   
I was reading up on it and it appears you can do something with rebase to eliminate them,  essentially moving your point forward,  but I've yet to figure it out.

Reply:
@station384 for your last bit, if you have no changes in your fork's `unstable` branch you can do:
```
git reset --hard upstream/unstable
git push --force
```

Then any time you branch off of your fork's unstable it won't have all the changes, as the head has been moved forward.
Unfortunately, you'll need to do this every time upstream (hey, that's me) updates.

Reply:
TYVM  that is exactly what I was looking for. 
It has been driving me nuts.

Now to make that command into a git alias :)